packages/truffle-core/lib/mnemonics/mnemonic.js

```diff
@@ -0,0 +1,73 @@
+/**
+* @module mnemonic;
+* @requires module:Configstore
+* @requires module:random
+* @requires module:bip39
+* @requires module:ethereumjs-wallet/hdkey
+*/
+
+const Configstore = require('configstore');
+const seedrandom = require('seedrandom');
+const bip39 = require('bip39'); 
+const hdkey = require('ethereumjs-wallet/hdkey');
+const crypto = require('crypto');
+
+const mnemonic = {
+
+  /**
+  * gets user-level mnemonic from user config, and if missing generates a new mnemonic
+  * @returns {String} mnemonic
+  */
+  mnemonic: function () {
```

Comment:
Just an idea but what about naming this something like `getOrGenerateMnemonic`?  It's a bit longer but it might be more descriptive.

Reply:
Oh sorry, I just meant for the function defined on line 21.